models.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+from shadja import db
+'''
+Contains all models
+'''
+
+subscription_pincode = db.Table('subscriptions_pincodes',
+    db.Column('subscription_id', db.Integer, db.ForeignKey('subscriptions.id')),
+    db.Column('pincode_code', db.Integer, db.ForeignKey('pincodes.code'))
+)
+
+class Pincode(db.Model):
+    __tablename__ = 'pincodes'
+    '''Class to represent every new pincode added to system by users'''
+    code = db.Column(db.Integer, primary_key=True)
+    availabilities = db.Column(db.JSON)
+    # hash of all availabilities, using Session.hash_many()
+    availabilities_hash = db.Column(db.String(24))
+    subscrptions = db.relationship(
+        "Subscription", 
+        secondary=subscription_pincode,
+        lazy=True, 
+        backref=db.backref("pincodes", lazy=True))
+
+    def __init__(self, code, availabilities=json.dumps({})):
+        self.code = code
+        self.availabilities = availabilities
+
+    def __repr__(self):
+        return f'Pin<{self.code}>'
+
+
+class Subscription(db.Model):
+    __tablename__ = 'subscriptions'
+    '''Class object to represent every new _request_ added to the system by users'''
+    id = db.Column(db.Integer, primary_key=True)
+    old = db.Column(db.Boolean)
+    want_free = db.Column(db.Boolean)
+    flavor = db.Column(db.String(32))
```

Comment:
These two are optional (basically we need another bit for signifying "don't care"

Reply:
Hmm. None is don't care, no?

Reply:
How does `db` store None? Can a boolean variable in `db` be set to None?